xclim/core/units.py

```diff
@@ -412,16 +412,7 @@ def convert_units_to(
 
     # TODO remove backwards compatibility of int/float thresholds after v1.0 release
     if isinstance(source, (float, int)):
-        if context == "hydro":
-            source_unit = units.mm / units.day
-        else:
-            source_unit = units.degC
-        warnings.warn(
-            "Future versions of xclim will require explicit unit specifications.",
-            FutureWarning,
-            stacklevel=3,
-        )
-        return units.Quantity(source, units=source_unit).to(target_unit).m
+        raise DeprecationWarning("Please specify units explicitly.")
```

Comment:
"raise" a "Warning" ?

 ```suggestion
        raise TypeError("Please specify units explicitly.")
```

Also, FYI: `FutureWarning` is target to users, while `DeprecationWarning` is targeted to developers and is hidden by default. (https://docs.python.org/3/library/exceptions.html#DeprecationWarning)

Reply:
It's also not recommended that `Warnings` be raised; In the case of a `Warning` that doesn't unwind the program, the `warnings.warn()` function is a much better option (used in many places within xclim).